src/components/OrderBookWidget.tsx

```diff
@@ -1,31 +1,180 @@
+          //
+          // For asks, we can offset the "startLocation" by 0.5. However, Amcharts does not support a "startLocation" of -0.5.
+          // For bids, we therefore offset the curve by -1 (expose the previous total volume) and use an offset of 0.5.
+          // Otherwise our steps would be off by one.
+          askValueY: type == Offer.Bid ? previousTotalVolume : null,
+          bidValueY: type == Offer.Ask ? totalVolume : null,
+        }
+      })
+  )
+}
+
+const draw = (
+  chartElement: HTMLElement,
+  baseToken: TokenDetails,
+  quoteToken: TokenDetails,
+  dataSource: string,
+): am4charts.XYChart => {
+  am4core.useTheme(am4themesSpiritedaway)
+  const chart = am4core.create(chartElement, am4charts.XYChart)
+
+  // Add data
+  chart.dataSource.url = dataSource
+  chart.dataSource.adapter.add('parsedData', function(data) {
+    data = JSON.parse(data)
+    const processed = processData(data.bids, baseToken, quoteToken, Offer.Bid).concat(
+      processData(data.asks, baseToken, quoteToken, Offer.Ask),
+    )
+    processed.sort((lhs, rhs) => lhs.price - rhs.price)
+    return processed
+  })
+
+  // Set up precision for numbers
+  chart.numberFormatter.numberFormat = '#,###.##'
+
+  // Create axes
+  const xAxis = chart.xAxes.push(new am4charts.CategoryAxis())
+  xAxis.dataFields.category = 'price'
+  xAxis.title.text = `Price (${baseToken.symbol}/${quoteToken.symbol})`
+
+  const yAxis = chart.yAxes.push(new am4charts.ValueAxis())
+  yAxis.title.text = 'Volume'
+
+  // Create series
+  const bidCurve = chart.series.push(new am4charts.StepLineSeries())
+  bidCurve.dataFields.categoryX = 'price'
+  bidCurve.dataFields.valueY = 'askValueY'
+  bidCurve.strokeWidth = 2
+  bidCurve.stroke = am4core.color('#0f0')
+  bidCurve.fill = bidCurve.stroke
+  bidCurve.startLocation = 0.5
+  bidCurve.fillOpacity = 0.1
+  bidCurve.tooltipText = 'Bid: [bold]{categoryX}[/]\nTotal volume: [bold]{totalVolume}[/]\nVolume: [bold]{volume}[/]'
+
+  const askCurve = chart.series.push(new am4charts.StepLineSeries())
+  askCurve.dataFields.categoryX = 'price'
+  askCurve.dataFields.valueY = 'bidValueY'
+  askCurve.strokeWidth = 2
+  askCurve.stroke = am4core.color('#f00')
+  askCurve.fill = askCurve.stroke
+  askCurve.fillOpacity = 0.1
+  askCurve.startLocation = 0.5
+  askCurve.tooltipText = 'Ask: [bold]{categoryX}[/]\nTotal volume: [bold]{totalVolume}[/]\nVolume: [bold]{volume}[/]'
+
+  const series3 = chart.series.push(new am4charts.ColumnSeries())
+  series3.dataFields.categoryX = 'price'
+  series3.dataFields.valueY = 'volume'
+  series3.strokeWidth = 0
+  series3.fill = am4core.color('#000')
+  series3.fillOpacity = 0.2
+
+  // Add cursor
+  chart.cursor = new am4charts.XYCursor()
+  return chart
+}
+
 const OrderBookWidget: React.FC<OrderBookProps> = props => {
-  const { baseToken, quoteToken } = props
+  const { baseToken, quoteToken, networkId } = props
+  const mountPoint = useRef<HTMLDivElement>(null)
+
+  useEffect(() => {
```

Comment:
well done! I see you go the hang of it fast :) 
also, very well the `useRef`

Reply:
@Velenir did that for me 😄